examples/chart/teleport-demo/README.md

```diff
@@ -0,0 +1,106 @@
+# Teleport on Kubernetes
+
+[Gravitational Teleport](https://github.com/gravitational/teleport) is a modern SSH/Kubernetes API proxy server for
+remotely accessing clusters of Linux containers and servers via SSH, HTTPS, or Kubernetes API.
+
+This configuration is quite a Gravitational-specific deployment but should show a good amount of reusability for other
+savvy admins.
+
+## Introduction
+
+This chart deploys Teleport components to your cluster using various Kubernetes primitives.
+
+It supports a few key features:
+- A configurable number of nodes per cluster (n)
+- One 'main' cluster with <n> nodes in its own Kubernetes namespace
+- Any amount of different-named trusted clusters with <n> nodes, each in their own Kubernetes namespace
+    - These clusters are automatically linked to 'main' as trusted clusters
+- OIDC authentication via Auth0
+- DNS records pointing to a Kubernetes LoadBalancer for each cluster, set up on a configurable Cloudflare domain
+- LetsEncrypt certificates automatically provisioned, configured and renewed for each cluster via certbot-dns-cloudflare
+- Secrets encrypted using sops and a key from GKE
+
+See the comments in the default `values.yaml` and also the [Teleport documentation](https://gravitational.com/teleport/docs/quickstart) for more options.
+
+## Prerequisites
+
+- Kubernetes 1.10+
+- [sops](https://github.com/mozilla/sops)
+- [helm-secrets](https://github.com/futuresimple/helm-secrets)
+- [gcloud SDK](https://cloud.google.com/sdk/docs/downloads-interactive)
+    - ```curl https://sdk.cloud.google.com | bash``` for a simple install
+- Secrets stored in secrets.yaml and encrypted with sops
+    - Teleport Enterprise license
+    - Email address and API key for a Cloudflare account that controls the domain you wish to use
+    - Client ID and client secret for a configured Auth0 application
+
+## Installing the chart
+
+If you want to use a different version of Teleport, you should build and push the Docker images for the specified
+version to GCR:
+
+```
+$ cd examples/chart/teleport-demo/docker
+$ gcloud auth login
+$ gcloud auth configure-docker
+$ ./build-all.sh 3.1.7
+```
+
+Make sure that you have access to the key for sops encryption:
+```bash
+$ gcloud auth application-default login
+$ gcloud kms keys list --location global --keyring teleport-sops
+NAME                                                                                          PURPOSE          LABELS  PRIMARY_ID  PRIMARY_STATE
+projects/kubeadm-167321/locations/global/keyRings/teleport-sops/cryptoKeys/teleport-sops-key  ENCRYPT_DECRYPT          1           ENABLED
+```
+
+kubectl needs to know about your cluster - for GKE you can use something like this:
+
+```bash
+$ gcloud container clusters get-credentials <cluster-name> --zone <zone> --project <project>
+$ ./gke-init.sh
+```
+
+Make sure that you have updated the submodule containing the secrets:
+
+```bash
+$ git pull --recurse-submodules
```

Comment:
Yeah - this is an unfortunate side effect. I think that overall, this code serves best as as a reference deployment because it has a lot of Gravitational-specific stuff in (such as the requirement to log in with Auth0, use of Cloudflare for DNS etc) and could probably work for someone else with some minor changes.

I'm not sure if @klizhentas has thoughts on whether I should try and make this much more generic so that it has immediate value for people outside of Gravitational who just want a multi-cluster deployment with Helm?

Reply:
its good for now